src/integer_mod_q/poly_over_zq/arithmetic/add.rs

```diff
@@ -0,0 +1,187 @@
+
+    /// testing addition for two [`PolyOverZq`]
+    #[test]
+    fn add() {
+        let a: PolyOverZq = PolyOverZq::from_str("3  2 4 1 mod 7").unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 7").unwrap();
+        let c: PolyOverZq = a + b;
+        assert_eq!(c, PolyOverZq::from_str("3  4 1 2 mod 7").unwrap());
+    }
+
+    /// testing addition for two borrowed [`PolyOverZq`]
+    #[test]
+    fn add_borrow() {
+        let a: PolyOverZq = PolyOverZq::from_str("3  2 4 1 mod 7").unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 7").unwrap();
+        let c: PolyOverZq = &a + &b;
+        assert_eq!(c, PolyOverZq::from_str("3  4 1 2 mod 7").unwrap());
+    }
+
+    /// testing addition for borrowed [`PolyOverZq`] and [`PolyOverZq`]
+    #[test]
+    fn add_first_borrowed() {
+        let a: PolyOverZq = PolyOverZq::from_str("3  2 4 1 mod 7").unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 7").unwrap();
+        let c: PolyOverZq = &a + b;
+        assert_eq!(c, PolyOverZq::from_str("3  4 1 2 mod 7").unwrap());
+    }
+
+    /// testing addition for [`PolyOverZq`] and borrowed [`PolyOverZq`]
+    #[test]
+    fn add_second_borrowed() {
+        let a: PolyOverZq = PolyOverZq::from_str("3  2 4 1 mod 7").unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 7").unwrap();
+        let c: PolyOverZq = a + &b;
+        assert_eq!(c, PolyOverZq::from_str("3  4 1 2 mod 7").unwrap());
+    }
+
+    /// testing addition for big [`PolyOverZq`]
+    #[test]
+    fn add_large_numbers() {
+        let a: PolyOverZq = PolyOverZq::from_str(&format!(
+            "3  -{} 4 {} mod {}",
+            u64::MAX,
+            i64::MIN,
+            u64::MAX - 58
+        ))
+        .unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str(&format!(
+            "3  {} 4 {} mod {}",
+            i64::MIN,
+            i64::MIN,
+            u64::MAX - 58
+        ))
+        .unwrap();
+        let c: PolyOverZq = a + b;
+        assert!(
+            c == PolyOverZq::from_str(&format!(
+                "3  -{} 8 {} mod {}",
+                i128::from(i64::MAX) + 59,
+                -59,
+                u64::MAX - 58
+            ))
+            .unwrap()
+        );
+    }
+
+    /// testing addition for [`PolyOverZq`] with different moduli does not work
+    #[test]
+    #[should_panic]
+    fn add_mismatching_modulus() {
+        let a: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 7").unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 8").unwrap();
+        let _c: PolyOverZq = a + b;
+    }
+
+    /// testing whether add_safe throws an error for mismatching moduli
+    #[test]
+    fn add_safe_is_err() {
+        let a: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 7").unwrap();
+        let b: PolyOverZq = PolyOverZq::from_str("3  -5 4 1 mod 11").unwrap();
+        assert!(&a.add_safe(&b).is_err());
+    }
```

Comment:
please add a test where after addition the largest coefficient is zero. Please double check, that the length has been reduced